src/hub/permissions.py

```diff
@@ -28,9 +28,9 @@ class CensorHub(RuleBasedPermission):
 
     def satisfies_rule(self, request):
         if request.method == "DELETE":
-            return request.user.is_authenticated and request.user.moderator
+            return request.user.is_authenticated and request.user.moderator and request.user.is_anonymous is not True
```

Comment:
It's best to check if the user is anonymous first because if the user is anonymous, the fields `is_authenticated`  and `moderator` won't exist and will result in an error.

Reply:
Good point 👍